tests/manage/pv_services/test_rwo_pvc_fencing_unfencing.py

```diff
@@ -0,0 +1,1011 @@
+            nodes.restart_nodes_teardown()
+            try:
+                node.wait_for_nodes_status(status=constants.NODE_READY)
+            except ResourceWrongStatusException:
+                # Restart the nodes if in NotReady state
+                not_ready_nodes = [
+                    n for n in node.get_node_objs() if n
+                    .ocp.get_resource_status(n.name) == constants.NODE_NOT_READY
+                ]
+                if not_ready_nodes:
+                    logger.info(
+                        f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+                    )
+                    nodes.restart_nodes(not_ready_nodes)
+                    node.wait_for_nodes_status(status=constants.NODE_READY)
+
+            # Check ceph health
+            assert ceph_health_check(), f"Ceph cluster health is not OK"
+            logger.info("Ceph cluster health is OK")
+
+        request.addfinalizer(finalizer)
+
+    def identify_and_add_nodes(self, scenario, num_of_nodes):
+        """
+        Fetches info about the worker nodes and add nodes (if required)
+
+        Args:
+            scenario (str): Scenario of app pods running on OCS or dedicated nodes
+                (eg., 'colocated', 'dedicated')
+            num_of_nodes (int): number of nodes required for running test
+
+        Returns:
+            tuple: tuple containing:
+                list: list of OCS nodes name
+                list: list of non-OCS nodes name
+
+        """
+        nodes_to_add = 0
+        initial_worker_nodes = helpers.get_worker_nodes()
+        ocs_nodes = machine.get_labeled_nodes(constants.OPERATOR_NODE_LABEL)
+        non_ocs_nodes = list(set(initial_worker_nodes) - set(ocs_nodes))
+
+        if 'colocated' in scenario and len(ocs_nodes) < num_of_nodes:
+            nodes_to_add = num_of_nodes - len(initial_worker_nodes)
+
+        if 'dedicated' in scenario and len(non_ocs_nodes) < num_of_nodes:
+            nodes_to_add = num_of_nodes - len(non_ocs_nodes)
+
+        if nodes_to_add > 0:
+            logger.info(f"{nodes_to_add} extra workers nodes needed")
+            if config.ENV_DATA['deployment_type'] == 'ipi':
+                machine_name = machine.get_machine_from_node_name(
+                    random.choice(initial_worker_nodes)
+                )
+                machineset_name = machine.get_machineset_from_machine_name(
+                    machine_name
+                )
+                machineset_replica_count = machine.get_replica_count(
+                    machineset_name
+                )
+                machine.add_node(
+                    machineset_name,
+                    count=machineset_replica_count + nodes_to_add
+                )
+                logger.info("Waiting for the new node(s) to be in ready state")
+                machine.wait_for_new_node_to_be_ready(machineset_name)
+            else:
+                # TODO: Add required num of nodes, instead of skipping
```

Comment:
Please create an issue for this implementation and add the issue # here

Reply:
Ack, I'll add the issue #
The issue already exists, Ecosystem team is already working on it.